samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManagerUtil.java

```diff
@@ -411,4 +438,292 @@ public static Set<String> getSideInputStoreNames(
+
+    Map<String, Map<String, String>> oldStateCheckpointMarkers = oldCheckpoint.getStateCheckpointMarkers();
+
+    Map<String, Map<String, String>> newStateCheckpointMarkers = ImmutableMap.<String, Map<String, String>>builder()
+        .put(KafkaChangelogStateBackendFactory.class.getName(), oldStateCheckpointMarkers.get(KafkaChangelogStateBackendFactory.class.getName()))
+        .put(BlobStoreStateBackendFactory.class.getName(), newSCMs)
+        .build();
+
+    CheckpointV2 checkpointV2 = new CheckpointV2(checkpointId, checkpointOffsets, newStateCheckpointMarkers);
+    checkpointManager.writeCheckpoint(taskName, checkpointV2);
+    return checkpointV2;
+  }
+
+  private static void createCheckpointDirFromStoreDirCopy(TaskName taskName, TaskModel taskModel,
+      File loggedStoreBaseDir, Set<String> storeName, CheckpointId checkpointId) {
+    StorageManagerUtil storageManagerUtil = new StorageManagerUtil();
+    for (String store : storeName) {
+      try {
+        File storeDirectory =
+            storageManagerUtil.getTaskStoreDir(loggedStoreBaseDir, store, taskName, taskModel.getTaskMode());
+        File checkpointDir = new File(storageManagerUtil.getStoreCheckpointDir(storeDirectory, checkpointId));
+        FileUtils.copyDirectory(storeDirectory, checkpointDir);
+      } catch (IOException exception) {
+        String msg = String.format("Unable to create a copy of store directory %s into checkpoint dir %s while "
+            + "attempting to recover from DeletedException", store, checkpointId);
+        throw new SamzaException(msg, exception);
+      }
+    }
+  }
+
+  private static BlobStoreManager getBlobStoreManager(Config config, ExecutorService executor) {
+    BlobStoreConfig blobStoreConfig = new BlobStoreConfig(config);
+    String blobStoreManagerFactory = blobStoreConfig.getBlobStoreManagerFactory();
+    BlobStoreManagerFactory factory = ReflectionUtil.getObj(blobStoreManagerFactory, BlobStoreManagerFactory.class);
+    return factory.getRestoreBlobStoreManager(config, executor);
+  }
+
+  private static void updateRestoreTime(long startTime, SamzaContainerMetrics samzaContainerMetrics,
+      TaskName taskInstance) {
+    long timeToRestore = System.currentTimeMillis() - startTime;
+    if (samzaContainerMetrics != null) {
+      Gauge taskGauge = samzaContainerMetrics.taskStoreRestorationMetrics().getOrDefault(taskInstance, null);
+
+      if (taskGauge != null) {
+        taskGauge.set(timeToRestore);
+      }
+    }
+  }
+
+  private static Boolean isUnwrappedExceptionDeletedException(Throwable ex) {
+    Throwable unwrappedException = FutureUtil.unwrapExceptions(CompletionException.class,
+        FutureUtil.unwrapExceptions(SamzaException.class, ex));
+    return unwrappedException instanceof DeletedException;
+  }
+
+  private static Metadata createSnapshotMetadataRequest(TaskName taskName, JobConfig jobConfig, String store) {
```

Comment:
Typo? What request is being created?

Reply:
This creates a request metadata. Updated the name to clarify.